packages/api/src/models/medical/tcm-encounter.ts

```diff
@@ -0,0 +1,55 @@
+import { DataTypes, Sequelize } from "sequelize";
+import { TcmEncounter } from "../../domain/medical/tcm-encounter";
+import { BaseModel, ModelSetup } from "../_default";
+
+export type TcmEncounterCreation = Omit<TcmEncounter, "id" | "createdAt" | "updatedAt" | "eTag">;
+
+export class TcmEncounterModel extends BaseModel<TcmEncounterModel> implements TcmEncounter {
+  static NAME = "tcm_encounter";
+  declare cxId: string;
+  declare patientId: string;
+  declare facilityName: string;
+  declare latestEvent: "Admitted" | "Transferred" | "Discharged";
+  declare class: string;
+  declare admitTime: Date | null;
+  declare dischargeTime: Date | null;
+  declare clinicalInformation: Record<string, unknown>;
+
+  static setup: ModelSetup = (sequelize: Sequelize) => {
+    TcmEncounterModel.init(
+      {
+        ...BaseModel.attributes(),
+        cxId: {
+          type: DataTypes.UUID,
+        },
+        patientId: {
+          type: DataTypes.UUID,
+        },
+        facilityName: {
+          type: DataTypes.STRING,
+        },
+        latestEvent: {
+          type: DataTypes.ENUM("Admitted", "Transferred", "Discharged"),
+        },
+        class: {
+          type: DataTypes.STRING,
+        },
+        admitTime: {
+          type: DataTypes.DATE,
+          allowNull: true,
```

Comment:
these (`allowNull`) we can leave to the migration only. We only need types here

Reply:
if we get the nullability right on the model as well we end up getting better type inference, right? You're right though, columns are nullable by default. I updated some columns to have allowNull:false instead